### PR TITLE
test: swap/quote, swap/tokens and swap/prices tests

### PR DESCRIPTION
**Pull request by fragosti**
# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [![Version](https://img.shields.io/github/package-json/v/0xProject/0x-api)](https://github.com/0xProject/0x-api/releases)
 [![Docs](https://img.shields.io/badge/docs-website-yellow.svg)](https://0x.org/docs/api)
 [![Chat with us on Discord](https://img.shields.io/badge/chat-Discord-blueViolet.svg)](https://discord.com/invite/d3FTX3M)
-[![Continuous Integration](https://github.com/0xProject/0x-api/workflows/Build%20and%20Test/badge.svg)](https://circleci.com/gh/0xProject/0x-api/tree/master)
+[![Continuous Integration](https://github.com/0xProject/0x-api/workflows/Build%20and%20Test/badge.svg)](https://github.com/0xProject/0x-api/actions?query=workflow%3A%22Build+and+Test%22+branch%3Amaster)
```
Comment (xianny) [inline README.md:4]: dang, I changed the badge but forgot to change the actual link! thanks for catching this.

https://github.com/0xProject/0x-api/pull/213/files#diff-04c6e90faac2675aa89e2176d2eec7d8R4

File: test/swap_test.ts [hunk 2]
```diff
@@ -1,39 +1,303 @@
...
     });
     after(async () => {
+        await blockchainLifecycle.revertAsync();
+        await teardownMeshAsync(SUITE_NAME);
         await teardownApiAsync(SUITE_NAME);
     });
-
-    describe('should respond to GET /swap/quote', () => {
+    describe('/quote', () => {
         it("with INSUFFICIENT_ASSET_LIQUIDITY when there's no liquidity (empty orderbook, sampling excluded, no RFQ)", async () => {
```
Comment (xianny) [inline test/swap_test.ts:103]: nit: can we change to "should response with..." for readability

File: test/swap_test.ts [hunk 3]
```diff
@@ -1,39 +1,303 @@
...
+        it('should respect buyAmount', async () => {
+            await quoteAndExpectAsync({ buyAmount: '1234' }, { buyAmount: '1234' });
+        });
+        it('should respect sellAmount', async () => {
+            await quoteAndExpectAsync({ sellAmount: '1234' }, { sellAmount: '1234' });
+        });
+        it('should respect gasPrice', async () => {
+            await quoteAndExpectAsync({ sellAmount: '1234', gasPrice: '150000000000' }, { gasPrice: '150000000000' });
+        });
+        it('should respect exludedSources', async () => {
```
Comment (xianny) [inline test/swap_test.ts:149]: nit (typo): exludedSources -> excludedSources

File: test/swap_test.ts (hunk 2 above)
Comment (fragosti) [reply test/swap_test.ts:103]: well this is describing the end-point its testing.

**Comment by fragosti** [comment]
@xianny thanks for the review. I would rather not have all the test cases defined in array. It would look good for now but makes it awkward when you actually want to code something up. I expect the test cases we add to prevent regressions will be more edge-casey than this.

File: README.md (hunk 1 above)
Comment (fragosti) [reply README.md:4]: Yea honestly I just needed a diff to get rid of the circle CI "failed" build and found this :P
